src/goto-instrument/code_contracts.cpp

```diff
@@ -155,6 +155,15 @@ void code_contractst::check_apply_loop_contracts(
     invariant = conjunction(invariant.operands());
   }
 
+  // Vector representing a (possibly multidimensional) decreases clause
+  exprt::operandst decreases_clause_vector = decreases_clause.operands();
+  int size_decreases_clasue = decreases_clause_vector.size();
```

Comment:
I think this should be `size_t` not `int`, but please double check what STL says. Not that we would have vectors that are longer than INT_MAX, but it's always good to use the correct type :)

```suggestion
  const size_t size_decreases_clasue = decreases_clause_vector.size();
```

Also use `size_t` in place of `int` for the loop counters below.

Reply:
Interestingly, there is a recent Linux exploit that used a `size_t` to `int` overflow: https://www.qualys.com/2021/07/20/cve-2021-33909/sequoia-local-privilege-escalation-linux.txt
😃